lib/Runtime/HostManager/HostManager.cpp

```diff
@@ -448,3 +455,59 @@ HostManager::runNetwork(llvm::StringRef networkName,
   activeRequestCount_--;
   return currentRun;
 }
+
+/// Helper to get the parameters in DeviceConfig from \p str. The \p str has
+/// multiple lines, and each line with this format : "str1" : "str2".
+static llvm::StringMap<std::string> getBackendParams(std::string &str) {
+  llvm::StringMap<std::string> ret{};
+  std::string s;
+  std::istringstream f(str.c_str());
+  while (getline(f, s, '\n')) {
+    // Abstract the mapping from each line's string:
+    // ""str1" : "str2"" => ret["str1"] = "str2";
+    size_t pos1, pos2, pos3, pos4;
+    pos1 = s.find('"');
+    assert(pos1 != std::string::npos && "invalid string format");
+    pos2 = s.find('"', pos1 + 1);
+    assert(pos2 != std::string::npos && "invalid string format");
+    pos3 = s.find('"', pos2 + 1);
+    assert(pos3 != std::string::npos && "invalid string format");
+    pos4 = s.find('"', pos3 + 1);
+    assert(pos4 != std::string::npos && "invalid string format");
+    ret[s.substr(pos1 + 1, pos2 - pos1 - 1)] =
+        s.substr(pos3 + 1, pos4 - pos3 - 1);
+  }
+  return ret;
+}
+
+/// If the device config file \p loadDeviceDoncfigsFile available, load \p
+/// configs from the file. Otherwise, create \p numDevices number of devices
+/// based on \p backendName.
+std::vector<std::unique_ptr<runtime::DeviceConfig>>
+runtime::generateDeviceConfigs(unsigned int numDevices,
+                               llvm::StringRef backendName, size_t memSize) {
+  std::vector<std::unique_ptr<runtime::DeviceConfig>> configs;
+  if (loadDeviceConfigsFileOpt.empty()) {
```

Comment:
I kind of like you now explicitly use loadDeviceConfigsFileOpt  in this function and do not take it as a parameter anymore. But it also makes it impossible to use this function if a profile-file-name comes from somewhere else rather then from this option. Is it a problem?

Reply:
I don't think so. It might need to be refactored down the road if we load all options from a config file, but for now I think using the option explicitly should be fine.